pandas/_libs/tslibs/src/datetime/np_datetime.c

```diff
@@ -498,6 +498,25 @@ npy_datetime npy_datetimestruct_to_datetime(NPY_DATETIMEUNIT base,
     return ret;
 }
 
+/*
+ * Computes the python `ret, d = divmod(d, unit)`.
+ *
+ * Note that GCC is smart enough at -O2 to eliminate the `if(*d < 0)` branch
+ * for subsequent calls to this command - it is able to deduce that `*d >= 0`.
+ */
```

Comment:
Can you add to this comment pointing back to the numpy PR.

Reply:
just added it. Thanks! 😄